src/main/java/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,24 @@
+package strategy;
+
+import java.util.HashMap;
+import java.util.Map;
+import model.Fruit;
+import service.OperationHandler;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private Map<String, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl() {
+        this.operationHandlerMap = new HashMap<>();
+    }
+
+    @Override
+    public void addOperation(String type, OperationHandler operation) {
+        operationHandlerMap.put(type, operation);
+    }
+
+    @Override
+    public void operationCheck(String type, Fruit key, String value) {
+        operationHandlerMap.get(type).doOperation(key, value);
+    }

```

Comment:
краще мати який проміжний тип, щоб не передавати сюди три параметри. Адже в мабйтньому кількість цих параметрів може лише зростати:
```suggestion
    public void operationCheck(FruitTransaction fruitTransaction) {
        operationHandlerMap.get(fruitTransaction.getOperationType()).doOperation(fruitTransaction);
    }
```